### PR TITLE
bootstrap: do not purge docs on CI environment

### LINKED ISSUE #117430: CI: Bootstrap no longer documents compiler crates
Bootstrap no longer documents any `rustc_*` / `compiler/` crates or `rustdoc` in CI.
[The index](https://doc.rust-lang.org/nightly/nightly-rustc/index.html) now only contains a meager 20 or so crates.
This has to have regressed fairly recently.

### PR DESCRIPTION
**Pull request by onur-ozkan**
This is a temporary fix for #117430, for more information please read https://github.com/rust-lang/rust/issues/117430#issuecomment-1788160523.

Fixes #117430

### CODE REVIEW

**Comment by onur-ozkan** [comment]
r? Mark-Simulacrum

File: src/bootstrap/src/core/builder.rs [hunk 1]
```diff
@@ -1273,7 +1274,7 @@ impl<'a> Builder<'a> {
             self.clear_if_dirty(&out_dir, &backend);
         }
 
-        if cmd == "doc" || cmd == "rustdoc" {
+        if cmd == "doc" || cmd == "rustdoc" && !CiEnv::is_ci() {
```
Comment (onur-ozkan) [inline src/bootstrap/src/core/builder.rs:1277]: I think this shouldn't cause any problem? I assume that the runners are automatically purging the entire build related state upon the termination.

**Comment by onur-ozkan** [comment]
As this PR resolves quite important bug, please feel free to review and r+ if changes looks good.

**Comment by RalfJung** [comment]
@bors p=1
we should get the nightly rustc docs fixed ASAP

**Comment by onur-ozkan** [comment]
> error parsing HTTP 429 response body: invalid character 'T' looking for beginning of value: "Too Many Requests (HAP429).\n"

@bors retry p=1

**Comment by onur-ozkan** [comment]
>   Step 1/56 : FROM ubuntu:22.04
  Head "https://registry-1.docker.io/v2/library/ubuntu/manifests/22.04": error parsing HTTP 429 response body: invalid character 'T' looking for beginning of value: "Too Many Requests (HAP429).\n"

Again!? lol..

@bors retry

**Comment by RalfJung** [comment]
This looks systematic, let's close the tree.
@bors treeclosed=100

Cc @rust-lang/infra

**Comment by RalfJung** [comment]
Hm, but strangely no other PR seems affected?
both [this](https://github.com/rust-lang-ci/rust/actions/runs/6736168858/job/18310932712) and [this](https://github.com/rust-lang-ci/rust/actions/runs/6735906397/job/18310132250) didn't have this issue (they failed, but elsewhere and later), and https://github.com/rust-lang/rust/pull/117521 also seems fine. And https://www.dockerstatus.com/ claims that everything is fine on the docker side.

**Comment by onur-ozkan** [comment]
I have no idea, let me rebase the latest upstream and let's give another shot then?

**Comment by RalfJung** [comment]
Now I am also seeing it in [other jobs](https://github.com/rust-lang/rust/actions/runs/6737561605/job/18315278867). Seems to be an on-and-off thing.

**Comment by RalfJung** [comment]
@onur-ozkan so you are saying this did [not help](https://github.com/rust-lang/rust/issues/117430#issuecomment-1791609163)? Should we revert it?

**Comment by onur-ozkan** [comment]
> @onur-ozkan so you are saying this did [not help](https://github.com/rust-lang/rust/issues/117430#issuecomment-1791609163)? Should we revert it?

Yes, we should.

**Comment by pnkfelix** [comment]
* single improvement to syn-opt-full that is almost certainly noise.
* and was reverted anyway in PR #117533

**Pull request merged**